crates/ide-completion/src/completions/pattern.rs

```diff
@@ -154,7 +154,6 @@ pub(crate) fn complete_pattern_path(
                     }
 
                     ctx.iterate_path_candidates(&ty, |item| match item {
-                        AssocItem::TypeAlias(ta) => acc.add_type_alias(ctx, ta),
```

Comment:
Why is this gone now?

Reply:
Without removing this, a couple of snapshot tests fail because completions for type aliases are emitted several times / duplicated. I figured this code was a workaround the fact that `method_resolution.rs` did not consider `TypeAlias`es, which changed in this PR, so I removed these explicit iterations of `TypeAlias`es.

Same for all your other comments for changes in `src/completions/*`. This is the main thing I'm uncertain about in this PR; it looks like adding support for `TypeAlias`es had impacts in other parts of the code (which led me to make these changes, hence the question "should be only handle `TypeAlias`es when resolving doc comments" above.

With that said, it looks like handling `TypeAlias`es in a generic way in `method_resolution.rs` allows us to explicitly handle them in other parts of the codebase, so this may be a good thing. I am not aware of _why_ it was handled differently before, though; it's possible that there were strong reasons for handling it differently which this PR breaks (subtly enough that additional tests do not fail).

Reply:
For instance if I undo this particular diff, I get the following test failure:

```
---- tests::pattern::enum_qualified stdout ----


error: expect test failed
   --> crates/ide-completion/src/tests/pattern.rs:314:9

You can update all `expect!` tests by running:

    env UPDATE_EXPECT=1 cargo test

To update a single test, place the cursor on `expect` token and use `run` feature of rust-analyzer.

Expect:
----
ct ASSOC_CONST const ASSOC_CONST: ()
bn RecordV {…} RecordV { field$1 }$0
bn TupleV(…)   TupleV($1)$0
bn UnitV       UnitV$0

----

Actual:
----
ct ASSOC_CONST const ASSOC_CONST: ()
ta AssocType   type AssocType = ()
bn RecordV {…} RecordV { field$1 }$0
bn TupleV(…)   TupleV($1)$0
bn UnitV       UnitV$0

----

Diff:
----
ct ASSOC_CONST const ASSOC_CONST: ()
ta AssocType   type AssocType = ()    (NOTE: THIS LINE WAS ADDED)
bn RecordV {…} RecordV { field$1 }$0
bn TupleV(…)   TupleV($1)$0
bn UnitV       UnitV$0

----
```

For this particular case it looks like the problem is that the `AssocType` is wrongly suggested, but I do remember that for other changes there were duplicate completions.